python/tvm/relay/qnn/op/legalizations.py

```diff
@@ -22,10 +22,25 @@
 from tvm import relay
 from .. import op as reg
 
-# Registering QNN Conv2D legalization function.
-@reg.register_qnn_legalize("qnn.conv2d")
-def legalize_qnn_conv2d(attrs, inputs, types):
-    """Legalizes QNN conv2d op.
+def shift_zero_point(data, out_dtype):
+    """Shifts (add/subtracts) the qnn tensor with +/-128)"""
+    if out_dtype == 'uint8':
+        shift = 128
+    elif out_dtype == 'int8':
+        shift = -128
+    else:
+        raise ValueError("Unsupport out dtype.")
+    data_modified = relay.cast(data, 'int32')
+    data_modified = relay.add(data_modified, relay.const(shift, 'int32'))
+    data_modified = relay.cast(data_modified, out_dtype)
+    return data_modified
+
+
+def arm_cpu_legalization(attrs, inputs, types, op):
```

Comment:
i would rather not name a generic arm_cpu_leagalization as it indicates it will do it for all ops. but here we are saying we can do it for ops that have a data and kernel part. maybe we should name this method more specific.